runtime/src/lib.rs

```diff
@@ -272,15 +272,15 @@ pub trait Listener: Sync + Send + 'static {
 /// messages over a network connection.
 pub trait Sink: Sync + Send + 'static {
     /// Send a message to the sink.
-    fn send(&mut self, msg: &[u8]) -> impl Future<Output = Result<(), Error>> + Send;
+    fn send<B: BoundedBuf>(&mut self, msg: B) -> impl Future<Output = Result<(), Error>> + Send;
```

Comment:
Don't love the naming of `IoBuf`/`BoundedBuf` but not sure what to replace it with yet. Putting here as a placeholder.